### ISSUE TITLE
Rework/redisign cloudinit iso-generation 

### ISSUE THREAD

**Issue by MalloZup** [dmacvicar/terraform-provider-libvirt#465]
So this is issue is for improving the `cloudinit` design.

We find out currently that requiring `mkisofs` is not universal enough in different os ( some version of SLES/ and some debian lack on this pkg).
Using also `geniso` was the same issue.

This issue is more about thinking on how we could redisign the iso generation part in more scalable/universal way.

I was thinking to use `usb` disk and `dd`. I think this could be possible and more universal.

But maybe other ways are possible

**Comment by moio** [comment]
How about using https://github.com/kdomanski/iso9660, do you see any immediate downside or roadblock?

**Comment by MalloZup** [comment]
good point. I think i didn't check it but is a good link to look at. thx for re-opining :)

**Comment by CyberDomovoy** [comment]
For debian:

```sh
apt update && apt install xorriso && update-alternatives --install /usr/bin/mkisofs mkisofs /usr/bin/xorrisofs 10
````

**Comment by bigon** [comment]
In the mean time https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=680949 has been fixed in debian unstable/testing so installing the `genisoimage` package should be enough

**Comment by MalloZup** [comment]
thx

**Comment by kstieger** [comment]
Hi,
we create our config isos with https://github.com/diskfs/go-diskfs

Example: https://gist.github.com/kstieger/66399edf56ca967b9ac232e6c438c4cd

**Comment by NHAS** [comment]
Ah okay, terraform 1.5.0 is broken, terraform 1.4.6 is totally fine. This isnt an issue for here I'll open one directly.

**Comment by scabala** [comment]
Related to #1018

**Comment by dmacvicar** [comment]
ℹ️ ℹ️ ℹ️ ℹ️ ℹ️ ℹ️

This issue report is relevant to the legacy version of the provider, which is now in the v0.8 branch of this repository.

Future development is based on a new provider, which is not compatible with this one, nor does share code.

As the new provider solves many issues with the legacy, and to make development in my free time more enjoyable, I have decided to close all bugs for the legacy provider, and to ask to check if the bug would apply to the new one. This also to encourage trying the new version.

- https://registry.terraform.io/providers/dmacvicar/libvirt/latest

and check the documentation:

- https://github.com/dmacvicar/terraform-provider-libvirt/blob/main/README.md
- https://registry.terraform.io/providers/dmacvicar/libvirt/latest/docs

You are free to reopen the bug for v0.8. It may encourage someone to fix it. My efforts will go into the new provider.

I ask you to check the new provider and re-evaluate this bug. 🙏
